### PR TITLE
Add `supplier` and `manufacturer` image link in `smarty url helper`

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add `supplier` and `manufacturer` image link in `smarty url helper`.
| Type?             | improvement 
| Category?         | FO 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | See description below 👇🏿
| Possible impacts? | N/A

## How to test

- Add a new `supplier` and a new `manufacturer` (with logo please 😉)
- edit `/themes/classic/templates/_partials/header.tpl` or another theme `.tpl` file and add : 

```html
   <img src='{url entity=supplierImage id=MY_NEW_SUPPLIER_ID_HERE}'>
   <img src='{url entity=manufacturerImage id=MY_NEW_MANUFACTURER_ID_HERE}'>
```
- now see your supplier and manufacturer images ;)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25543)

### CODE REVIEW

File: classes/Link.php [hunk 1]
```diff
@@ -1527,6 +1527,20 @@ public static function getUrlSmarty($params)
...
 
+                break;
+            case 'manufacturerImage':
+                $link = $context->link->getManufacturerImageLink(
+                    (int) $params['id'],
+                    $params['type'] = (isset($params['type']) ? $params['type'] : null)
+                );
+
+                break;
+            case 'supplierImageLink':
```
Comment (Progi1984) [inline classes/Link.php]:
```suggestion
            case 'supplierImage':
```

**Review by Progi1984 — COMMENTED** [review]
Hi, thank you for your contribution. Could you change the case for being consistent ?

File: classes/Link.php (hunk 1 above)
Comment (PululuK) [reply classes/Link.php]: Eagle eyes ;)

**Review by Progi1984 — APPROVED** [review]

File: classes/Link.php [hunk 2]
```diff
@@ -1527,6 +1527,20 @@ public static function getUrlSmarty($params)
                     $params['type'] = (isset($params['type']) ? $params['type'] : null)
                 );
 
+                break;
+            case 'manufacturerImage':
+                $link = $context->link->getManufacturerImageLink(
+                    (int) $params['id'],
+                    $params['type'] = (isset($params['type']) ? $params['type'] : null)
```
Comment (PierreRambaud) [inline classes/Link.php:1534]: It still weird to see this param attribution everywhere :sweat_smile:

**Review by PierreRambaud — APPROVED** [review]

File: classes/Link.php (hunk 2 above)
Comment (PululuK) [reply classes/Link.php:1534]: I totally agree ... I'm wondering if i could do something here or open another PR?

Comment (PierreRambaud) [reply classes/Link.php:1534]: Maybe in another PR, it will be easier as this one has already been approved.

**Comment by hibatallahAouadni** [comment]
Hello @PululuK 

Thanks for your PR :clap: 
Without PR:

image

With PR: 

- No image or the Id is wrong => display the default image :heavy_check_mark: 
image

- Id is right => :heavy_check_mark: 

image

LGTM, QA :heavy_check_mark: 
Thanks!

**Comment by PululuK** [comment]
Thank you @hibatallahAouadni

**Pull request merged**